src/utils/form-helper.test.ts

```diff
@@ -80,4 +95,404 @@ describe('Form Engine Helper', () => {
+        expect(field.fieldDependants).toBeUndefined();
+      });
+    });
+
+    it('should not overwrite existing fieldDependants', () => {
+      const field: FormField = {
+        label: 'Field with Validator',
+        type: 'obs',
+        questionOptions: {
+          rendering: 'number',
+        },
+        id: 'field-1',
+        validators: [
+          {
+            type: 'conditionalAnswered',
+            referenceQuestionId: 'field-2',
+          },
+        ],
+      };
+
+      const referencedField: FormField = {
+        label: 'Referenced Field',
+        type: 'obs',
+        questionOptions: {
+          rendering: 'number',
+        },
+        id: 'field-2',
+        fieldDependants: new Set(['field-3']),
+      };
+
+      const allFields: FormField[] = [field, referencedField];
+
+      evaluateConditionalAnswered(field, allFields);
+
+      expect(referencedField.fieldDependants).toEqual(new Set(['field-3', 'field-1']));
+    });
+  });
+
+  describe('evaluateFieldReadonlyProp', () => {
+    it('should not change field.readonly if it is not empty', () => {
+      const field: FormField = {
+        label: 'Test Field',
+        type: 'obs',
+        questionOptions: {
+          rendering: 'number',
+        },
+        id: 'field-1',
+        readonly: true,
+      };
+
+      evaluateFieldReadonlyProp(field, false, false, false);
+
+      expect(field.readonly).toBe(true);
+    });
+
+    it('should set field.readonly to true if sectionReadonly is not empty', () => {
+      const field: FormField = {
+        label: 'Test Field',
+        type: 'obs',
+        questionOptions: {
+          rendering: 'number',
+        },
+        id: 'field-1',
+        readonly: '',
+      };
+
+      evaluateFieldReadonlyProp(field, 'some value', false, false);
+
+      expect(field.readonly).toBe(true);
+    });
+
+    it('should set field.readonly to true if pageReadonly is not empty', () => {
+      const field: FormField = {
+        label: 'Test Field',
+        type: 'obs',
+        questionOptions: {
+          rendering: 'number',
+        },
+        id: 'field-1',
+        readonly: '',
+      };
+
+      evaluateFieldReadonlyProp(field, false, 'some value', false);
+
+      expect(field.readonly).toBe(true);
+    });
+
+    it('should set field.readonly to true if formReadonly is true', () => {
+      const field: FormField = {
+        label: 'Test Field',
+        type: 'obs',
+        questionOptions: {
+          rendering: 'number',
+        },
+        id: 'field-1',
+        readonly: '',
+      };
+
+      evaluateFieldReadonlyProp(field, false, false, true);
+
+      expect(field.readonly).toBe(true);
+    });
+  });
+
+  describe('parseToLocalDateTime', () => {
+    it('should parse valid date string with time correctly', () => {
+      const dateString = '2023-06-27T14:30:00';
+      const expectedDate = new Date(2023, 5, 27, 14, 30, 0);
+      const parsedDate = parseToLocalDateTime(dateString);
```

Comment:
@samuelmale why aren't we using [parseDate](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/docs/API.md#parsedate)?

Reply:
No good reason! I agree we should rely more on the utils exposed by the framework.